crypto/bn/bn_gf2m.c

```diff
@@ -336,6 +336,8 @@ int BN_GF2m_mod_arr(BIGNUM *r, const BIGNUM *a, const int p[])
         z[j] = 0;
 
         for (k = 1; p[k] != 0; k++) {
+            if (p[k] < 0 || p[k] >= p[k - 1])
```

Comment:
Well these are public API functions, so I don't think we can change the way you are supposed to interact with them now.